src/App.tsx

```
@@ -1,61 +1,129 @@
+}
+
+interface Todo {
+  id: number;
+  title: string;
+  userId: number;
+  completed: boolean;
+  user: User;
+}
 
 export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState('');
+  const [errors, setErrors] = useState({ title: false, user: false });
+
+  useEffect(() => {
+    setUsers(usersFromServer);
+    setTodos(
+      todosFromServer.map(todo => ({
+        ...todo,
+        user: usersFromServer.find(user => user.id === todo.userId)!,
+      })),
+    );
+  }, []);
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const hasTitle = title.trim() !== '';
+    const hasUser = userId !== '';
+
+    setErrors({ title: !hasTitle, user: !hasUser });
+    if (!hasTitle || !hasUser) {
+      return;
+    }
+
+    const newId = todos.length ? Math.max(...todos.map(t => t.id)) + 1 : 1;
+    const user = users.find(u => u.id === Number(userId));
+
+    if (!user) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: newId,
+      title,
+      userId: user.id,
+      completed: false,
+      user,
+    };
+
+    setTodos(prev => [...prev, newTodo]);
+    setTitle('');
+    setUserId('');
+    setErrors({ title: false, user: false });
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">Title</label>
+          <input
+            type="text"
+            data-cy="titleInput"
+            value={title}
+            onChange={event => {
+              setTitle(event.target.value);
+
+              if (errors.title && event.target.value.trim() !== '') {
```

Validation clearing for the title error currently only clears when the user types a non-empty value: `if (errors.title && event.target.value.trim() !== '') { ... }`. The task requires: "Hide the message immediately after any change of the field with an error". Change this so the title error is cleared on any change (i.e. remove the `trim() !== ''` condition) so the error is hidden immediately when the user types, even if the field becomes empty again. Reference: task description on error behavior.